test/unit/test_devices/test_ios_device.py

```diff
@@ -341,6 +341,45 @@ def test_enable_from_config(self):
         self.device.native.check_config_mode.assert_called()
         self.device.native.exit_config_mode.assert_called()
 
+    @mock.patch.object(IOSDevice, "_image_booted", side_effect=[False, True])
+    @mock.patch.object(IOSDevice, "set_boot_options")
+    @mock.patch.object(IOSDevice, "reboot")
+    @mock.patch.object(IOSDevice, "_wait_for_device_reboot")
+    def test_install_os_image_booted_false_true(self, mock_wait, mock_reboot, mock_set_boot, mock_image_booted):
+        state=self.device.install_os(BOOT_IMAGE)
+        mock_set_boot.assert_called()
+        mock_reboot.assert_called()
+        mock_wait.assert_called()
+        self.assertEqual(state, True)
+
+    @mock.patch.object(IOSDevice, "_image_booted", side_effect=[True, True])
+    @mock.patch.object(IOSDevice, "set_boot_options")
+    @mock.patch.object(IOSDevice, "reboot")
+    @mock.patch.object(IOSDevice, "_wait_for_device_reboot")
+    def test_install_os_image_booted_true_true(self, mock_wait, mock_reboot, mock_set_boot, mock_image_booted):
+        state=self.device.install_os(BOOT_IMAGE)
+        mock_set_boot.assert_not_called()
+        mock_reboot.assert_not_called()
+        mock_wait.assert_not_called()
+        self.assertEqual(state, False)
+
+    @mock.patch.object(IOSDevice, "_image_booted", side_effect=[False, False])
+    @mock.patch.object(IOSDevice, "set_boot_options")
+    @mock.patch.object(IOSDevice, "reboot")
+    @mock.patch.object(IOSDevice, "_wait_for_device_reboot")
+    def test_install_os_image_booted_false_false(self, mock_wait, mock_reboot, mock_set_boot, mock_image_booted):
+        self.assertRaises(OSInstallError, self.device.install_os, BOOT_IMAGE)
+        
+    @mock.patch.object(IOSDevice, "_image_booted", side_effect=[True, False])
+    @mock.patch.object(IOSDevice, "set_boot_options")
+    @mock.patch.object(IOSDevice, "reboot")
+    @mock.patch.object(IOSDevice, "_wait_for_device_reboot")
+    def test_install_os_image_booted_true_false(self, mock_wait, mock_reboot, mock_set_boot, mock_image_booted):
+        state=self.device.install_os(BOOT_IMAGE)
+        mock_set_boot.assert_not_called()
+        mock_reboot.assert_not_called()
+        mock_wait.assert_not_called()
+        self.assertEqual(state, False)
```

Comment:
This is the same test as `test_os_install_already_installed` and should be removed

Reply:
Fixed